steering_controllers/src/steering_controllers.cpp

```diff
@@ -0,0 +1,489 @@
+  }
+  const auto age_of_last_command = get_node()->now() - msg->header.stamp;
+
+  if (ref_timeout_ == rclcpp::Duration::from_seconds(0) || age_of_last_command <= ref_timeout_)
+  {
+    input_ref_.writeFromNonRT(msg);
+  }
+  else
+  {
+    RCLCPP_ERROR(
+      get_node()->get_logger(),
+      "Received message has timestamp %.10f older for %.10f which is more then allowed timeout "
+      "(%.4f).",
+      rclcpp::Time(msg->header.stamp).seconds(), age_of_last_command.seconds(),
+      ref_timeout_.seconds());
+  }
+}
+
+void SteeringControllers::reference_callback_unstamped(
+  const std::shared_ptr<geometry_msgs::msg::Twist> msg)
+{
+  auto twist_stamped = *(input_ref_.readFromNonRT());
+  twist_stamped->header.stamp = get_node()->now();
+  // if no timestamp provided use current time for command timestamp
+  if (twist_stamped->header.stamp.sec == 0 && twist_stamped->header.stamp.nanosec == 0u)
+  {
+    RCLCPP_WARN(
+      get_node()->get_logger(),
+      "Timestamp in header is missing, using current time as command timestamp.");
+    twist_stamped->header.stamp = get_node()->now();
+  }
+
+  const auto age_of_last_command = get_node()->now() - twist_stamped->header.stamp;
+
+  if (ref_timeout_ == rclcpp::Duration::from_seconds(0) || age_of_last_command <= ref_timeout_)
+  {
+    twist_stamped->twist = *msg;
+  }
+  else
+  {
+    RCLCPP_ERROR(
+      get_node()->get_logger(),
+      "Received message has timestamp %.10f older for %.10f which is more then allowed timeout "
+      "(%.4f).",
+      rclcpp::Time(twist_stamped->header.stamp).seconds(), age_of_last_command.seconds(),
+      ref_timeout_.seconds());
+  }
+}
+
+controller_interface::InterfaceConfiguration SteeringControllers::command_interface_configuration()
+  const
+{
+  controller_interface::InterfaceConfiguration command_interfaces_config;
+  command_interfaces_config.type = controller_interface::interface_configuration_type::INDIVIDUAL;
+  command_interfaces_config.names.reserve(nr_cmd_itfs_);
+
+  for (size_t i = 0; i < params_.rear_wheel_names.size(); i++)
+  {
+    command_interfaces_config.names.push_back(
+      params_.rear_wheel_names[i] + "/" + hardware_interface::HW_IF_VELOCITY);
+  }
+  for (size_t i = 0; i < params_.front_steer_names.size(); i++)
+  {
+    command_interfaces_config.names.push_back(
+      params_.front_steer_names[i] + "/" + hardware_interface::HW_IF_POSITION);
+  }
+  return command_interfaces_config;
+}
+
+controller_interface::InterfaceConfiguration SteeringControllers::state_interface_configuration()
+  const
+{
+  controller_interface::InterfaceConfiguration state_interfaces_config;
+  state_interfaces_config.type = controller_interface::interface_configuration_type::INDIVIDUAL;
+
+  state_interfaces_config.names.reserve(nr_state_itfs_);
+  const auto rear_wheel_feedback = params_.position_feedback ? hardware_interface::HW_IF_POSITION
+                                                             : hardware_interface::HW_IF_VELOCITY;
```

Comment:
This feedback decision could be additionally made by flags in parameters:

- `front_axis_steer`
- `rear_axis_steer`